base/sort.jl

```diff
@@ -711,31 +918,19 @@ function sort!(v::AbstractVector;
                by=identity,
                rev::Union{Bool,Nothing}=nothing,
                order::Ordering=Forward)
-    ordr = ord(lt,by,rev,order)
-    if (ordr === Forward || ordr === Reverse) && eltype(v)<:Integer
-        n = length(v)
-        if n > 1
-            min, max = extrema(v)
-            (diff, o1) = sub_with_overflow(max, min)
-            (rangelen, o2) = add_with_overflow(diff, oneunit(diff))
```

Comment:
Why don't you use these functions when computing similar quantities in your new code?

Reply:
See https://github.com/JuliaLang/julia/pull/44230/#discussion_r825364777

Note that the method I use dispatches to count sort on `rand(Int8, 1000)`, but the old method overflows and so fails to dispatch to counting sort.